x-pack/plugins/security_solution/public/flyout/entity_details/shared/components/insight_entity/misconfigurations_overview.tsx

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback } from 'react';
+import { css } from '@emotion/react';
+import { EuiFlexGroup, EuiFlexItem, EuiText, useEuiTheme } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { DistributionBar } from '@kbn/security-solution-distribution-bar';
+import { useMisconfigurationPreview } from '@kbn/cloud-security-posture';
+import { euiThemeVars } from '@kbn/ui-theme';
+import { i18n } from '@kbn/i18n';
+import { ExpandablePanel } from '../../../../shared/components/expandable_panel';
+
+export const MisconfigurationsOverview = ({ hostName }: { hostName: string }) => {
+  const queryHostName = useCallback(() => {
```

Comment:
This `useCallback` doesn't make much sense. You are always calling the function on line 38. So, it created a new object for every render. If you want to memoise the query between render to prevent unnecessary rerender, use `useMemo`.

```suggestion
  const hostNameQuery = useMemo(() => {
```

And update line 38 to use `hostNameQuery`


It also might be a good idea to move the query-building function to another file so we don't mix UI and ES querying logic in the same file.

Reply:
@machadoum 
woops, i forgot to close this PR but basically we will split this PR into smaller PR like this https://github.com/elastic/kibana/pull/190933 , 
please leave the review on this other PR :) Thanks !